services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -458,6 +482,91 @@ private void handleStoreStateLookup(ChannelHandlerContext ctx, VenicePathParserH
+      return;
+    }
+
+    int currentVersionNumber = store.getCurrentVersion();
+    if (currentVersionNumber == Store.NON_EXISTING_VERSION) {
+      setupResponseAndFlush(BAD_REQUEST, REQUEST_TOPIC_ERROR_NO_CURRENT_VERSION.getBytes(), false, ctx);
+      return;
+    }
+
+    Optional<Version> currentVersionOptional = store.getVersion(currentVersionNumber);
+    if (!currentVersionOptional.isPresent()) {
+      setupResponseAndFlush(INTERNAL_SERVER_ERROR, REQUEST_TOPIC_ERROR_MISSING_CURRENT_VERSION.getBytes(), false, ctx);
+      return;
+    }
+
+    final HybridStoreConfig hybridStoreConfig;
+    Version currentVersion = currentVersionOptional.get();
+    if (currentVersion.isUseVersionLevelHybridConfig()) {
+      if (currentVersion.getHybridStoreConfig() == null) {
+        setupResponseAndFlush(BAD_REQUEST, REQUEST_TOPIC_ERROR_CURRENT_VERSION_NOT_HYBRID.getBytes(), false, ctx);
+        return;
+      }
+      hybridStoreConfig = currentVersion.getHybridStoreConfig();
+    } else {
+      hybridStoreConfig = store.getHybridStoreConfig();
+    }
+
+    /**
+     * Only allow router request_topic for hybrid stores that have data replication policy:
+     * 1. NON_AGGREGATE
+     * 2. ACTIVE_ACTIVE
+     */
+    DataReplicationPolicy dataReplicationPolicy = hybridStoreConfig.getDataReplicationPolicy();
+    if (!dataReplicationPolicy.equals(NON_AGGREGATE) && !dataReplicationPolicy.equals(ACTIVE_ACTIVE)) {
+      setupResponseAndFlush(BAD_REQUEST, REQUEST_TOPIC_ERROR_UNSUPPORTED_REPLICATION_POLICY.getBytes(), false, ctx);
+      return;
+    }
+
+    // Retrieve partitioner config from the store
+    PartitionerConfig storePartitionerConfig = store.getPartitionerConfig();
+    Map<String, String> queryParams = helper.extractQueryParameters(request);
+    if (queryParams.get(PARTITIONERS) != null) {
+      // Retrieve provided partitioner class list from the request
+      boolean hasMatchedPartitioner = false;
+      for (String partitioner: queryParams.get(PARTITIONERS).split(",")) {
+        if (partitioner.equals(storePartitionerConfig.getPartitionerClass())) {
+          hasMatchedPartitioner = true;
+          break;
+        }
+      }
+      if (!hasMatchedPartitioner) {
+        String errorMsg = String
+            .format(REQUEST_TOPIC_ERROR_FORMAT_UNSUPPORTED_PARTITIONER, storePartitionerConfig.getPartitionerClass());
+        setupResponseAndFlush(BAD_REQUEST, errorMsg.getBytes(), false, ctx);
+        return;
+      }
+    }
+
+    VersionCreationResponse responseObject = new VersionCreationResponse();
+    responseObject.setCluster(clusterName);
+    responseObject.setName(storeName);
+    responseObject.setPartitions(currentVersion.getPartitionCount());
+    responseObject.setKafkaTopic(Version.composeRealTimeTopic(storeName));
+    // RT topic only supports NO_OP compression
+    responseObject.setCompressionStrategy(CompressionStrategy.NO_OP);
```

Comment:
This is for RT >VT,  we do support compression there?

Reply:
The clients will only write to RT. We don't support compression in RT

Reply:
For hybrid stores, Leaders do the compression when producing to VT